Mage.Sets/src/mage/cards/i/IndominusRexAlpha.java

```diff
@@ -0,0 +1,183 @@
+    }
+}
+
+// Based on MindMaggotsEffect
+class IndominusRexAlphaCountersEffect extends OneShotEffect {
+
+    private static final List<CounterType> copyableCounters = Arrays.asList(
+            CounterType.FLYING,
+            CounterType.FIRST_STRIKE,
+            CounterType.DOUBLE_STRIKE,
+            CounterType.DEATHTOUCH,
+            CounterType.HEXPROOF,
+            CounterType.HASTE,
+            CounterType.INDESTRUCTIBLE,
+            CounterType.LIFELINK,
+            CounterType.MENACE,
+            CounterType.REACH,
+            CounterType.TRAMPLE,
+            CounterType.VIGILANCE
+    );
+
+    IndominusRexAlphaCountersEffect() {
+        super(Outcome.BoostCreature);
+        this.staticText = "discard any number of creature cards. It enters with a " + getFirstChoiceAsString() +
+                " counter on it if a card discarded this way has " + getFirstChoiceAsString()
+                + ". The same is true for " + getOtherChoicesAsString() + ".";
+        for (CounterType counterType : copyableCounters) {
+            Counter counter = counterType.createInstance();
+            if (!(counter instanceof AbilityCounter)) {
+                throw new IllegalArgumentException(counterType + " is not an ability counter");
+            }
+        }
+    }
+
+    protected String getFirstChoiceAsString(){
+        return copyableCounters.toArray(new CounterType[0])[0].getName();
+    }
+
+    protected String getOtherChoicesAsString(){
+        StringBuilder sb = new StringBuilder();
+        CounterType[] choicesArray = copyableCounters.toArray(new CounterType[0]);
+
+        // Skip first counter name - flying
+        for (int i = 1; i < copyableCounters.size(); i++){
+            String separator = ", ";
+            if (i == copyableCounters.size()-2){
+                separator = ", and ";
+            } else if (i == copyableCounters.size()-1){
+                separator = "";
+            }
+            sb.append(choicesArray[i].getName()).append(separator);
+        }
+        return sb.toString();
+    }
+
+    private IndominusRexAlphaCountersEffect(final IndominusRexAlphaCountersEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public IndominusRexAlphaCountersEffect copy() {
+        return new IndominusRexAlphaCountersEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller == null) {
+            return false;
+        }
+
+        // EntersBattlefieldAbility is static, see AddCountersSourceEffect
+        Permanent permanent = game.getPermanentEntering(source.getSourceId());
+        if (permanent == null) {
+            return true;
+        }
+
+        TargetCard target = new TargetDiscard(0, Integer.MAX_VALUE, StaticFilters.FILTER_CARD_CREATURE, controller.getId());
+        controller.choose(outcome, controller.getHand(), target, source, game);
```

Comment:
Looks like to AI bug with wrong outcome for choose dialog.

Current code uses ability default (boostcreature) — it’s “good” outcome so ai will try to choose max amount (discard max possible cards). Same problem in original maggot card. Make sure AI keep hand cards while casting it. If it discard whole hand then it buggy and must be fixed.

AI hint code can help here but it’s hard to manually check all cards for counters (custom ai logic is not good practice). So I recommends to use bad outcome here like outcome.AiDontUseIt with comments like “ai can’t choose cards, so ignore it to keep whole hand”.